### PR TITLE
Topics default open graph image

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Adds a default open graph image for all topic pages and allows overrides for specific topics by setting `primary_image`.

### Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-topic-open-graph/topics/agile)

### Solution

Updated logic in `head.html` to set the `primary_image` field from the default topic page.

**Default Site OG Image**

<details><summary>digitalgov-2022-card-v3</summary>
Screen Shot 2023-11-21 at 4 16 38 PM
</details>

**Default Topic OG Image**

<details><summary>topic-og-primary-image</summary>
Screen Shot 2023-11-21 at 4 18 04 PM
</details>

### How To Test

1. Visit any [topic page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-topic-open-graph/topics/agile)
2. Use https://www.opengraph.xyz to preview open graph image

### What To Test

1. All topics will display default topic OG image
2. Pick any topic and set `primary_image` to "10x-gold-x-logo" to override the default

---

### Dev Checklist

- [ ] PR has correct labels
- [ ] A11y testing (voice over testing, meets WCAG, run axe tools)
- [ ] Code is formatted properly

### CODE REVIEW

File: content/topics/_index.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
 title: "Topics"
 summary: ""
+primary_image: "topics-og-primary-image"
```
Comment (nick-mon1) [inline content/topics/_index.md:4]: `head.html` looks here to set the default topic image.

File: content/topics/design/_index.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ title: "Design"
 deck: "Understand how and why design impacts user experience"
 
 summary: "Guidance, resources, and community to help you use design to create government websites that meet customer needs, work well on any device, and follow federal web requirements."
-primary_image: "topics-og-primary-image"
```
Comment (nick-mon1) [inline content/topics/design/_index.md:12]: Removed in favor of using the new code.

File: themes/digital.gov/layouts/partials/core/head.html [hunk 3]
```diff
@@ -12,6 +12,16 @@
...
+{{/* Set primary_image from topics/_index.md if not set on individual topic */}}
+{{ if eq .Type "topics" }}
+  {{ if .Params.primary_image }}
+    {{ $primary_image = .Params.primary_image }}
+  {{ else }}
+    {{ $topics_page := .Site.GetPage "topics" }}
+    {{ $primary_image = $topics_page.Params.primary_image }}
+  {{ end }}
+{{ end }}
+
```
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/head.html:24]: Is this necessary? The same param is getting checked on lines 5-12.

**Review by mejiaj — COMMENTED** [review]
Question about possible duplicate logic, otherwise LGTM.

**Review by clmedders — APPROVED** [review]
looks good outside of James comment

File: themes/digital.gov/layouts/partials/core/head.html (hunk 3 above)
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/head.html:24]: I tried refactoring this code to check it once, but to set the correct default topics image and any topic specific override, we need to to check it within the `if eq .Type "topics"` "scope" to ensure the correct image is set.

Comment (mejiaj) [reply themes/digital.gov/layouts/partials/core/head.html:24]: I understand now. Thanks for clearing that up.

**Review by mejiaj — APPROVED** [review]

**Review by mejiaj — APPROVED** [review]

**Review by ToniBonittoGSA — APPROVED** [review]

**Pull request merged**